packages/terminal/src/browser/terminal-widget.ts

```diff
@@ -270,6 +284,16 @@ export class TerminalWidget extends BaseWidget {
         ));
     }
 
+    dispose(): void {
+
+        /* Close the backend terminal only when explicitly closting the terminal
+         * a refresh for example won't close it.  */
+        if (this.closeOnDispose === true && this.terminalId !== undefined) {
+            this.shellTerminalServer.close(this.terminalId);
+        }
```

Comment:
did you check whether `dispose` is actually called when closing the window?

Reply:
Yes it is, I tested closing the window closes the backend terminal , refresh diposes of everyting but the backend term.